File: Java Practical file/Q8.java
// Q8. WAP to print all prime no between 1 to 100.

class Q8 {
    // This method returns true if the given number is prime, else false
    public static boolean isPrime(int num) {
        if (num == 1 || num <= 0) {
            return false;
        } else {
            for (int i = 2; i <= num / 2; i++) {
                if (num % i == 0) {
                    return false;
                }
            }
        }
        return true;
    }

    public static void main(String[] args) {
        // Print a message indicating that we are finding prime numbers between 1 and
        // 100
        System.out.println("Prime no. b/w 1 to 100");
        // Loop through numbers 1 to 100 and check if they are prime
        for (int i = 1; i < 100; i++) {
            if (isPrime(i)) {
                // If the number is prime, print it to the console
                System.out.println(i + " is a Prime no.");
            }
        }
    }
}

File: Python pratical file/Q8.py
# Q8. Write a Python program to print all prime numbers in an interval.

# Get the lower and upper limits from the user
lower = int(input("Enter the lower limit: "))
upper = int(input("Enter the upper limit: "))

# Print a message indicating what the program is doing
print(f"Prime numbers between {lower} and {upper} are:")

# Loop through all numbers between the lower and upper limits, inclusive
for num in range(lower, upper+1):

    # Check if the current number is greater than 1, since 1 is not a prime number
    if num > 1:

        # Loop through all numbers between 2 and half the current number, inclusive
        # This is because any number greater than half of the current number cannot be a factor
        # that produces an integer quotient when divided into the current number
        for i in range(2, int(num/2)+1):

            # If the current number is divisible by i, then it is not a prime number
            if (num % i) == 0:

                # Exit the inner loop early if a divisor is found
                break

        # If the inner loop completes without finding a divisor, then the current number is prime
        else:

            # Print the prime number
            print(num, end=", ")

# Print a newline character to make the output look cleaner
print()
